docs/proposals/last-operation-description.md

```diff
@@ -0,0 +1,97 @@
+A PR is already in place to expose this information: https://github.com/openshift/ansible-service-broker/pull/546
+When called this apb module would update known annotations on the pod ie: ```apb_last_operation``` and ``` apb_final_operation```
+with the description provided by the apb developer. This would be part of the 
+[ansible playbook modules](https://github.com/ansibleplaybookbundle/ansible-asb-modules).
+
+In order to collect this information we would use a watch via the Kubernetes client on the pod resource within the temporary namespace [Pod Rest API](https://docs.openshift.com/container-platform/3.5/rest_api/kubernetes_v1.html#list-or-watch-objects-of-kind-pod).
+This would allow us to react to changes made (i.e to the annotations) on Pod Object. Whenever a change occurred, an update to the JobStatus would happen. If the ```apb_final_operation``` annotation is present this would take precedence over the last_operation annotation. 
+Once the pod was deleted we would stop the watch on the pod and update the JobStatus ```final_operation``` annotation value.
+
+Sudo code example
+
+```go
+
+wi, err := k8client.CoreV1().Pods(ns).Watch(meta_v1.ListOptions{})
+changes := wi.ResultChan()
+    for ch := range changes{
+		if ch.Type == watch.Modified{
+			...
+		}
+		if ch.Type == watch.Deleted{
+			close(ch)
+			...
+		}
+	}
+
+```
+
+As this would block, it would need to be done in a background go routine. Using a watch in a background routine should allow us to update the JobState independent of the actual execution of the apb.    
+
+
+### Broker changes
+Below are a set of changes that I believe are in line with the current design. The exact implementation would likely differ but the gist would be the same.
+
+- A new field would be added to the JobState type:
+```Description string ``` 
+The string value , added by the apb module and gathered from the pod annotation would be stored here.
+
+- Modify the existing provision and deprovision subscribers along with the corresponding work messages.
+A namespace field would be added to these messages as the pod name is already present. 
+Inside the subscriber, the go routine to watch the pod and update the JobStatus would start when a new work msg was recieved.
+This routine would stop once the a deleted change was received in the watch.
+
+- Add a new subscriber for bind
+async bind will be part of the service catalog, so having a Status and Description for an async binding will also be needed.
+Adding a new subscriber and workmsg for binding operations will allow us to update the JobStatus once async binding arrives.
+
+- Modify last operation handler to pull the description out of the Job state and send it back as part of the response.
+
+### APB changes
```

Comment:
Actually don't think that you are targeting the APB specifically here, more the underlying [ansible-asb-modules](https://github.com/ansibleplaybookbundle/ansible-asb-modules/blob/master/library/). I don't think that you would be putting the last operation on disk, you should just modify the running pod's annotation directly with the new `asb_last_operation_description`.